vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -46,8 +46,12 @@ def authentication_password_encrypted(type = nil)
     authentication_component(type, :password_encrypted)
   end
 
+  def required_credential_fields(_type)
```

Comment:
Well, `required_credential_fields` is new and not used anywhere else and all the current users expect `type = 'default'` now .. but the thing is, the actual required fields *may* depend on the type, so `has_credentials?` needs to pass the type there eventually, so why not do it properly now?